app/src/main/java/no/usn/mob3000/data/repository/auth/AuthRepository.kt

```diff
@@ -74,8 +74,25 @@ class AuthRepository(
      */
     override suspend fun logout() = supabase.auth.signOut()
 
-    override suspend fun register() {
-        TODO("Registration not yet implemented")
+    /**
+     * Registration with fetching of userdata.
+     *
+     * @author Anarox
+     * @created 2024-10-30
+     */
+    override suspend fun register(email: String, password: String): Result<DomainUser> {
+        return try {
+            supabase.auth.signUpWith(Email) {
+                this.email = email
+                this.password = password
+            }
+
+            currentUserId = authDataSource.getCurrentUserId()
+
+            Result.success(fetchUserData())
```

Comment:
Had to fix an SMTP issue, but now sign-up actually works again on the back-end. Anyway, that means I got time to test this.

This code will not work, that's why I told you to test what you get back from `signUpWith()`. It doesn't return a session at all, though it does return some data about the user that was created. E.g. the e-mail and so on and so forth. You'll need to rework how registration works. The only way to return the user would be to also call on the signin function inside the register function, but after `signUpWith`, but that does not really tickle my "that sounds like a good idea" bone, so we should find some other way first and compare the two.

Here's an object it returns, with the e-mail stripped.
```
UserInfo(appMetadata={"provider":"email","providers":["email"]}, aud=authenticated, confirmationSentAt=2024-10-31T07:33:06.071847984Z, confirmedAt=null, createdAt=2024-10-31T07:33:06.059134Z, email=[REDACTED], emailConfirmedAt=null, factors=[], id=d6dc8dad-0882-4901-ad54-9738816ad1bb, identities=[Identity(id=d6dc8dad-0882-4901-ad54-9738816ad1bb, identityData={"email":"[REDACTED]","email_verified":false,"phone_verified":false,"sub":"d6dc8dad-0882-4901-ad54-9738816ad1bb"}, identityId=68bd6759-8e9a-4b59-8df9-0402e41fecf5, lastSignInAt=2024-10-31T07:33:06.066021797Z, updatedAt=2024-10-31T07:33:06.06607Z, createdAt=2024-10-31T07:33:06.06607Z, provider=email, userId=d6dc8dad-0882-4901-ad54-9738816ad1bb)], lastSignInAt=null, phone=, role=authenticated, updatedAt=2024-10-31T07:33:07.566847Z, userMetadata={"email":"[REDACTED]","email_verified":false,"phone_verified":false,"sub":"d6dc8dad-0882-4901-ad54-9738816ad1bb"}, phoneChangeSentAt=null, newPhone=null, emailChangeSentAt=null, newEmail=null, invitedAt=null, recoverySentAt=null, phoneConfirmedAt=null, actionLink=null)
```